components/formats-gpl/src/loci/formats/in/FluoviewReader.java

```diff
@@ -304,12 +307,42 @@ protected void initStandardMetadata() throws FormatException, IOException {
       else if (name.equals("y")) {
         voxelY = voxel;
       }
-      else if (name.equals("z") || name.equals("event")) {
+      else if (name.equals("event")) {
         m.sizeZ *= size;
         if (dimensionOrder.indexOf("Z") == -1) {
           dimensionOrder += "Z";
         }
-        voxelZ = voxel;
+        if (Double.compare(voxelZ, 1) == 0) {
+          voxelZ = voxel;
+        }
+      }
+      else if (name.equals("z")) {
+        m.sizeZ *= size;
+        if (dimensionOrder.indexOf("Z") == -1) {
+          dimensionOrder += "Z";
+        }
+        
+        ArrayList<Double> uniqueZ = new ArrayList<Double>();
+        if (i > 1) {
+          double[] zPositions = stamps[i - 2];
+          for (Double z : zPositions) {
+            BigDecimal bd = new BigDecimal(z);
+            bd = bd.setScale(10, RoundingMode.HALF_UP);
+            if (!uniqueZ.contains(bd.doubleValue())) uniqueZ.add(bd.doubleValue());
+          }
+          }
+        if (uniqueZ.size() > 1 && uniqueZ.size() == size) {
+          BigDecimal lastZ = BigDecimal.valueOf(uniqueZ.get(uniqueZ.size() - 1));
+          BigDecimal firstZ = BigDecimal.valueOf(uniqueZ.get(0));
+          BigDecimal zRange = (lastZ.subtract(firstZ)).abs();
+          BigDecimal zSize = BigDecimal.valueOf((double)(getSizeZ() - 1));
+          voxelZ = zRange.divide(zSize).doubleValue();
```

Comment:
Note this line is causing some `ArithmeticException` for some sample files (see https://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-repository-subset/61). Excluding for now.